File: 94. Binary Tree Inorder Traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    List<Integer> output = new ArrayList<Integer>();
    public List<Integer> inorderTraversal(TreeNode root) {
        inorder(root);
        
        return output;
    }
    
    public void inorder(TreeNode root) {
        if (root != null) {
            inorder(root.left);
            output.add(root.val);
            inorder(root.right);
        }
    }
}

File: 94. Binary Tree Inorder Traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def inorderTraversal(self, root):
        """
        :type root: TreeNode
        :rtype: List[int]
        """
        def inorder(root):
            if root:
                inorder(root.left)
                output.append(root.val)
                inorder(root.right)
        
        output = []
        inorder(root)
        
        return output
